packages/x-data-grid-premium/src/components/promptControl/RecordButton.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from 'react';
+import { Timeout } from '@mui/utils/useTimeout';
+import useLazyRef from '@mui/utils/useLazyRef';
+import { useGridApiContext } from '../../hooks/utils/useGridApiContext';
+import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
+
+const BrowserSpeechRecognition = (globalThis as any).webkitSpeechRecognition;
```

Comment:
Does this button work for you?
I tried both Chrome and Safari, and both do not work. Chrome seems to start recording but immediately stops it.
`webkitSpeechRecognition` is available in both Chrome and Safari.

Reply:
Chrome works fine. Safari doesn't. It gets into recording state, but there is no query update while I am talking. Maybe it is a permissions issue. I will check it out

How is it working for you @KenanYusuf?

Reply:
I'm getting the same result as you @arminmeh. Chrome works. Safari doesn't, even if I set the microphone permissions to always allow 🤔

<img width="948" alt="Screenshot 2024-11-20 at 15 52 14" src="https://github.com/user-attachments/assets/092e1ec0-535d-4ee2-a1e3-1d4b1e34c347">

Reply:
Ok, Chrome works, it's Brave that doesn't work (it's Chromium though)

Reply:
OK, it won't work with Brave – even though the API is there, Google uses it's own backend services for transcription, which they offer for free in Chrome: https://github.com/brave/brave-browser/issues/3725